src/cmd/library_migrate.js

```diff
@@ -46,22 +43,19 @@ class AbstractLibraryMigrateCommand extends Command {
 	 *  - result: result of running `processLibrary()` if no errors were produced.
 	 *  - err: any error that was produced.
 	 */
-	run(state, site) {
-		const libsPromise = Promise.resolve(site.getLibraries());
-		return when.map(libsPromise, libdir => {
-			return Promise.resolve(site.notifyStart(libdir))
-			.then(() => {
-				const dir = path.resolve(libdir);
-				const repo = new FileSystemLibraryRepository(dir, FileSystemNamingStrategy.DIRECT);
-				return this.processLibrary(repo, '', state, site, libdir)
-				.then(([res, err]) => {
-					return Promise.resolve(site.notifyEnd(libdir, res, err))
-					.then(() => {
-						return { libdir, res, err };
-					});
-				});
-			});
-		});
+	async run(state, site) {
+		const libraries = await site.getLibraries();
```

Comment:
I confirmed that `Promise.resolve(promise)` returns that promise so this adaptation is correct.